src/Std/Data/DHashMap/RawDef.lean

```diff
@@ -0,0 +1,31 @@
+/-
+Copyright (c) 2018 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura, Mario Carneiro, Markus Himmel
+-/
+prelude
+import Std.Data.DHashMap.Internal.AssocList.Basic
+
+/-!
+# Definition of `DHashMap.Raw`
+
+This file defines the type `Std.Data.DHashMap.Raw`. All of its functions are defined in the module
+`Std.Data.DHashMap.Basic`.
+-/
+
+set_option linter.missingDocs true
+set_option autoImplicit false
+
+universe u v
+
+namespace Std.DHashMap
+
+/-- Dependent hash maps without bundled well-formedness invariant. Suitable for use in nested
+inductive types. The well-formedness invariant is called `Raw.WF`. -/
```

Comment:
```suggestion
/--
Dependent hash maps without a bundled well-formedness invariant, suitable for use in nested
inductive types. The well-formedness invariant is called `Raw.WF`. For most uses, the bundled
version `Std.DHashMap` is better.

This is a simple separate-chaining hash table. The data of the hash map consists of a cached
size and an array of buckets, where each bucket is an `AssocList α β` (which is the same as
a `List ((a : α) × β a)` but with one less level of indirection). The number of buckets is always a
power of two. The hash map doubles its size upon inserting an element such that the number of
elements is more than 75% of the number of buckets.
-/
```

Reply:
I'm not a fan of explicitly mentioning `AssocList` here, because that type is supposed to be an internal implementation detail. Would you be fine with "The data of the hash map consists of a cached
size and an array of buckets, where each bucket is a linked list of key-value pairs"?

Reply:
Absolutely. The thought is mostly to give readers a rough cost model, and the most important part of that is that it's based on an array.

Reply:
(and the second-most important part is the resizing/capacity strategy)